### ISSUE TITLE
Getting snapshot is slow

### ISSUE THREAD

**Issue by inoc603** [dgraph-io/dgraph#2197]
- Dgraph version
1.0.3
- Hardware
3 CentOS 7 instances with 16G RAM, 8 CPU, 10Gbps network connection
- Steps to reproduce the issue (command/config used to run Dgraph).
  1. Start zero on vpc1 with `--replicas 3`
  2. Use bulk loader on vpc1, and start a dgraph server with `--memory_mb 8192`
  3. Start dgraph servers on vpc2 and vpc3

- Expected behaviour and actual result.
The data in `rdf.gz` format is around 9MB, which is very small. I expect the snapshot to sync very quickly, but it takes around 4 minute for vpc2 and vpc3 to retrieve snapshot and become available, as shown in the server log on vpc2/3:

```
predicate.go:51: Getting SNAPSHOT: Time elapsed: 04m35s, bytes written: 349 MB
```

I checked disk IO, network speed, and CPU/memory usage on all instances. All seem normal. 

I was originally intended to use bulk loader to generate 3 shards, but I ran into the issue mentioned in #2129 . So I followed @pawanrawal 's suggestion and end up with this issue.

Any idea what is limiting the speed or where should I look for the cause of this problem?

Thanks.

**Comment by janardhan1993** [comment]
On first look, it seems sending is the bottleneck.  We will fix this in next release.

**Comment by janardhan1993** [comment]
@inoc603  Are you using hdd/sdd. Can you please share the commands you used to start dgraph instances(did you specify local/public address in `--my` flag)

**Comment by inoc603** [comment]
I'm using hdd on a vpc. I start dgraph zero and server with only a `--config` flag.

Zero config file:

```
replicas: 3
wal: /data/dgraph/zw
my: ${zero_ip}:5080
```

Server config file:

```
idx: 1
wal: /data/dgraph/w
postings: /data/dgraph/p
export: /data/dgraph/export
my: ${server_ip}:7080
memory_mb: 8192
zero: ${zero_ip}:5080
debugmode: true
```

I have 3 dgraph servers running, and they have different `idx` and ip address. I start them in serial because when starting in parallel, the servers sometimes don't join the cluster correctly.

**Comment by janardhan1993** [comment]
@inoc603 Dgraph is optimised to work on ssd and would be slower on hdd. 
We have done some changes to make it faster, can you please try using latest master.

**Comment by inoc603** [comment]
I'm seeing a big improvement on the latest master:

```
predicate.go:52: Getting SNAPSHOT: Time elapsed: 30s, bytes written: 332 MB, bytes/sec 11070537
```

FYI: I did some profiling on the vpc I ran the test. Using `iperf` between the vpcs where I ran zero and server, I get the max network speed is around `800MB/s`. Using `dd`  I get `80MB/s` disk writing speed. I'll try this on ssd later. I'm think it will be much faster.

After the fix, replicas are getting snapshot at around `10MB/s`, which will satisfy most of my usecases. I'll say that the fix is a good job done! Thanks!

Will this fix be included in the next release?
